app/models/crashlytics_integration.rb

```diff
@@ -0,0 +1,76 @@
+# == Schema Information
+#
+# Table name: crashlytics_integrations
+#
+#  id             :uuid             not null, primary key
+#  json_key       :string
+#  project_number :string
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#
+class CrashlyticsIntegration < ApplicationRecord
+  has_paper_trail
+
+  include Providable
+  include Displayable
+  include Firebasable
+  include Loggable
+
+  delegate :crashlytics_project, to: :config
+
+  API = Installations::Crashlytics::Api
+
+  RELEASE_TRANSFORMATIONS = {
+    new_errors_count: :new_errors_count,
+    errors_count: :errors_count,
+    sessions_in_last_day: :sessions_in_last_day,
+    sessions: :sessions,
+    sessions_with_errors: :sessions_with_errors,
+    daily_users_with_errors: :daily_users_with_errors,
+    daily_users: :daily_users,
+    total_sessions_in_last_day: :total_sessions_in_last_day,
+    external_release_id: :external_release_id
+  }
+
+  def installation
+    API.new(project_number, access_key)
+  end
+
+  def firebase_installation
+    Installations::Google::Firebase::Api.new(project_number, access_key)
+  end
+
+  def to_s
+    "crashlytics"
+  end
+
+  def find_release(platform, version, build_number)
+    return nil if version.blank?
+    installation.find_release(crashlytics_project(platform), version, build_number, RELEASE_TRANSFORMATIONS, integrable.bundle_identifier)
+  end
+
+  private
+
+  def bq_access?
+    data = installation.get_bq_data
+    data.present?
+  rescue Google::Cloud::PermissionDeniedError => ex
+    # TODO: this exception should be handled by the api layer with a crashlytics specific error and reason,
+    # and the reason should be checked here instead of a Google::Cloud::PermissionDeniedError directly
+    elog(ex)
+    nil
```

Comment:
**todo:** 

Previously, the bq_access? check was incorrect. We've fixed this up now. But there's some more work required here:

- We should handle the specific error mentioned above in the API layer and then we should decorate that as an "auth erorr", so that Google errors don't leak in the integration layer.
- Handle that decorated error here and log it.

Reply:
Thank you for the fix, we will work on the remaining changes